fxa/__main__.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import getpass
+import os
+import re
+import sys
+
+from fxa.errors import ClientError
+from fxa.tools.bearer import get_bearer_token
+from fxa.tools.browserid import get_browserid_assertion
+from fxa.tools.fxa import create_new_fxa_account
+
+FXA_API_URL = "https://api-accounts.stage.mozaws.net/v1"
+FXA_OAUTH_URL = "https://oauth.stage.mozaws.net/v1"
+DEFAULT_CLIENT_ID = "5882386c6d801776"  # Firefox dev Client ID
+
+
+def main(args=None):
+    """The main routine."""
+    if args is None:
+        args = sys.argv[1:]
+
+    parser = argparse.ArgumentParser(description="PyFxA commands")
+    parser.add_argument('--bearer',
+                        help='Generate a Bearer token',
+                        dest='bearer',
+                        action='store_true')
+
+    parser.add_argument('--browserid', '--bid',
+                        help='Generate a BrowserID assertion',
+                        dest='browserid',
+                        action='store_true')
+
+    parser.add_argument('--create', '-c',
+                        help='Create a new user',
+                        dest='create',
+                        action='store_true')
+    parser.add_argument('--auth', '-u',
+                        help='User credentials',
+                        dest='auth',
+                        required=False)
+    parser.add_argument('--out', '-o', '-O',
+                        help='Output file',
+                        dest='output_file',
+                        required=False,
+                        default=None)
+
+    parser.add_argument('--verbose', '-v',
+                        help='Display status',
+                        dest='verbose',
+                        action='store_true')
+
+    # Creation args
+    parser.add_argument('--user-salt',
+                        help=('Salt used to calculate the user credentials. '
+                              '(Random by default)'),
+                        dest='fxa_user_salt',
+                        required=False)
+
+    # FxA server configuration
+    parser.add_argument('--account-server',
+                        help='Firefox Account server URL',
+                        dest='account_server_url',
+                        required=False,
+                        default=FXA_API_URL)
+
+    parser.add_argument('--oauth-server',
+                        help='Firefox Account OAuth server URL',
+                        dest='oauth_server_url',
+                        required=False,
+                        default=FXA_OAUTH_URL)
```

Comment:
Over in fxa-oauth-client [1] there's a neat convention of using `--env=stage` or `--env=production` or similar to select between whole suites of URLs.  It could be a good addition here as well.  (Actually that lib has a `token` command very similar to what you're adding here, but it doesn't have a lot of the other stuff).

[1] https://github.com/mozilla/fxa-oauth-client/